### PR TITLE
NP-46527 Index document, supplement partOf

### PR DESCRIPTION
**Pull request by monaullah**
Jira issue: NP-46527 Index document, supplement partOf

For aggregation purposes, modify partOf-list to only contain identifiers (not whole URI): Example:
```
"contributors": [
  {
    "type": "NviContributor",
    "id": "https://example.org/person/997998",
    "affiliations": [
      {
        "type": "NviOrganization",
        "id": "https://example.org/organization/194.1.1.1”,
        "partOf": ["194.0.0.0”, "194.1.0.0", "194.1.1.0"]
        ]
      }
    ]
  }
]
```

Also, updated nvi context with the following:
```
"partOf": {
    "@container": "@set",
    "@context": {
      "@base": "__REPLACE_WITH_API_DOMAIN__/cristin/organization"
    },
    "@type": "@id"
  }
```

This change required an update of the OpenSearch `Query` `contributorQueryIncludingSubUnits`

### CODE REVIEW

**Review by brinxmat — COMMENTED** [review]
A bit weird, but if it is a requirement…

At the same time: https://json-ld.org/playground/#startTab=tab-expanded&json-ld=%7B%22%40context%22%3A%7B%22%40vocab%22%3A%22https%3A%2F%2Fexample.org%2Fontology%23%22%2C%22id%22%3A%22%40id%22%2C%22type%22%3A%22%40type%22%2C%22partOf%22%3A%7B%22%40context%22%3A%7B%22%40base%22%3A%22https%3A%2F%2Fexample.org%2Forganization%2F%22%7D%2C%22%40type%22%3A%22%40id%22%7D%7D%2C%22contributors%22%3A%5B%7B%22type%22%3A%22NviContributor%22%2C%22id%22%3A%22https%3A%2F%2Fexample.org%2Fperson%2F997998%22%2C%22affiliations%22%3A%5B%7B%22type%22%3A%22NviOrganization%22%2C%22id%22%3A%22https%3A%2F%2Fexample.org%2Forganization%2F194.1.1.1%22%2C%22partOf%22%3A%5B%22194.0.0.0%22%2C%22194.1.0.0%22%2C%22194.1.1.0%22%5D%7D%5D%7D%5D%7D

**Comment by monaullah** [comment]
> A bit weird, but if it is a requirement…
> 
> At the same time: https://json-ld.org/playground/#startTab=tab-expanded&json-ld=%7B%22%40context%22%3A%7B%22%40vocab%22%3A%22https%3A%2F%2Fexample.org%2Fontology%23%22%2C%22id%22%3A%22%40id%22%2C%22type%22%3A%22%40type%22%2C%22partOf%22%3A%7B%22%40context%22%3A%7B%22%40base%22%3A%22https%3A%2F%2Fexample.org%2Forganization%2F%22%7D%2C%22%40type%22%3A%22%40id%22%7D%7D%2C%22contributors%22%3A%5B%7B%22type%22%3A%22NviContributor%22%2C%22id%22%3A%22https%3A%2F%2Fexample.org%2Fperson%2F997998%22%2C%22affiliations%22%3A%5B%7B%22type%22%3A%22NviOrganization%22%2C%22id%22%3A%22https%3A%2F%2Fexample.org%2Forganization%2F194.1.1.1%22%2C%22partOf%22%3A%5B%22194.0.0.0%22%2C%22194.1.0.0%22%2C%22194.1.1.0%22%5D%7D%5D%7D%5D%7D

Good point, see latest commits

**Review by brinxmat — APPROVED** [review]

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java [hunk 1]
```diff
@@ -79,8 +80,8 @@ public CandidateQuery(CandidateQueryParameters params) {
 
     public static Query matchAtLeastOne(Query... queries) {
         return new Query.Builder()
-            .bool(new BoolQuery.Builder().should(Arrays.stream(queries).toList()).build())
-            .build();
+                   .bool(new BoolQuery.Builder().should(Arrays.stream(queries).toList()).build())
+                   .build();
```
Comment (StigNorland) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java:84]: Note that if you are looking or OR queries, this is way to do it in Opensearch...

```
QueryBuilders.disMaxQuery()
            .add(new TermQueryBuilder(field1, value))
           .add(new TermsQueryBuilder(field2, values))
           .....

```

Comment (StigNorland) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java:84]: hit on one or more of these Terms will include the document in the result

Comment (StigNorland) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java:84]: should isn't an OR query, it just boost queries that hits.

**Review by StigNorland — APPROVED** [review]

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java (hunk 1 above)
Comment (monaullah) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java:84]: Thanks for the tip. I haven't changed this code and the tests are green, so merging this PR. Can take a closer look at it and make a new PR with modifications.

Comment (StigNorland) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java:84]: This is only a problem when you have several keys to filter on, and you need something like this
`key=X & ( key2=Y || key2=Z ) & key3=BZ`

`key=X & ( key2=Y || key2=Z ) & key3=BZ || key4=CX`

what does the last OR mean?

**Pull request merged**